### PR TITLE
📝docs: clarify shuffle behavior and example in PyDataset

### PR DESCRIPTION
**Pull request by maitry63**
This PR updates the `PyDataset` class docstring inside `keras/src/trainers/data_adapters/py_dataset_adapter.py` to include the  documentation for the `shuffle´ argument.

### Context
Addresses issue #20142.

### Notes
- Adds clarification about `shuffle´  default behavior (False).
- Keeps the existing example intact, adding only the `shuffle´  usage line.
- No functional code changes.

### CODE REVIEW

File: keras/src/trainers/data_adapters/py_dataset_adapter.py [hunk 1]
```diff
@@ -66,10 +75,12 @@ class PyDataset:
 
     class CIFAR10PyDataset(keras.utils.PyDataset):
 
-        def __init__(self, x_set, y_set, batch_size, **kwargs):
+        def __init__(self, x_set, y_set, batch_size, shuffle=False, **kwargs):
             super().__init__(**kwargs)
             self.x, self.y = x_set, y_set
             self.batch_size = batch_size
+            self.shuffle = shuffle
+            self.indices = np.arange(len(self.x))
```
Comment (fchollet) [inline keras/src/trainers/data_adapters/py_dataset_adapter.py:84]: In the example, also do `np.random.shuffle(self.indices)` here when `shuffle` is True, since the best practice when shuffling is to do it for every epoch, not just epoch >= 1

File: keras/src/trainers/data_adapters/py_dataset_adapter.py [hunk 2]
```diff
@@ -40,6 +40,12 @@ class PyDataset:
             multiprocessed setting.
             Reduce this value to reduce the CPU memory consumption of
             your dataset. Defaults to 10.
+        shuffle: Whether to shuffle the sample ordering at the end of
+            each epoch. This argument is passed to `model.fit()`. When
+            `model.fit(..., shuffle=True)`, the training loop
```
Comment (fchollet) [inline keras/src/trainers/data_adapters/py_dataset_adapter.py]: Add verb: "When calling"

File: keras/src/trainers/data_adapters/py_dataset_adapter.py [hunk 3]
```diff
@@ -40,6 +40,12 @@ class PyDataset:
             multiprocessed setting.
             Reduce this value to reduce the CPU memory consumption of
             your dataset. Defaults to 10.
+        shuffle: Whether to shuffle the sample ordering at the end of
+            each epoch. This argument is passed to `model.fit()`. When
+            calling `model.fit(..., shuffle=True)`, the training loop
+            automatically calls `on_epoch_end()` at each epoch
+            boundary, allowing datasets to implement custom
+            shuffling logic. Defaults to `False`.
```
Comment (hertschuh) [inline keras/src/trainers/data_adapters/py_dataset_adapter.py:48]: The `shuffle` argument is not part of `PyDataset.__init__`, so it should be removed from here. It's just an argument in your example.

File: keras/src/trainers/data_adapters/py_dataset_adapter.py [hunk 4]
```diff
@@ -52,6 +58,9 @@ class PyDataset:
         over the dataset. They are not being used by the `PyDataset` class
         directly. When you are manually iterating over a `PyDataset`,
         no parallelism is applied.
+    - `shuffle=False` keeps the sample order fixed across epochs.
+        For distributed or deterministic training prefer
+        `shuffle=False` and manage the order externally.
```
Comment (hertschuh) [inline keras/src/trainers/data_adapters/py_dataset_adapter.py:63]: The `shuffle` argument is not part of `PyDataset.__init__`, so it should be removed from here. It's just an argument in your example.

File: keras/src/trainers/data_adapters/py_dataset_adapter.py [hunk 5]
```diff
@@ -66,10 +75,17 @@ class PyDataset:
 
     class CIFAR10PyDataset(keras.utils.PyDataset):
 
-        def __init__(self, x_set, y_set, batch_size, **kwargs):
+        def __init__(self, x_set, y_set, batch_size, shuffle=False, **kwargs):
```
Comment (hertschuh) [inline keras/src/trainers/data_adapters/py_dataset_adapter.py:78]: The `shuffle` argument is not part of `PyDataset.__init__`. I understand that you can do it this way here like in your example.

But in Keras, we add shuffling via `PyDatasetAdapter`. So it makes it confusing because if you do:
- `CIFAR10PyDataset(shuffle=True)` and `model.fit(shuffle=False)` it will be shuffled
- `CIFAR10PyDataset(shuffle=False)` and `model.fit(shuffle=True)` it will be shuffled

I guess my question is, what is the reason to add `shuffle` directly in the `PyDataset`?

Comment (brainbytes42) [reply keras/src/trainers/data_adapters/py_dataset_adapter.py:78]: Hi there,

I'm sorry, but I'm quite confused about sorting a PyDataset.

> But in Keras, we add shuffling via `PyDatasetAdapter`.

The PyDatasetAdapter seems to shuffle the order of the _batches_, see https://github.com/keras-team/keras/blob/df0ec013296ceb830de8ae79738e06cebeab2546/keras/src/trainers/data_adapters/py_dataset_adapter.py#L257-L261 but not the samples inside the batches - right?

So if I have samples [1, ..., 9] and batches of size 3 I'd get something like [4,5,6], [1,2,3], [7,8,9] if shuffeling via model.fit(...)?

I'd think to shuffle the _entire_ dataset, one would have to shuffle in `on_epoch_begin()` to have randomized samples from which the batches are generated? Like [8,3,6], [9,2,1], [5,4,7]. Or do I miss something that `model.fit(shuffle=True)` would do this? Is there any best practice to get all samples shuffled?

Thank you for clarification / advice on this.

Comment (hertschuh) [reply keras/src/trainers/data_adapters/py_dataset_adapter.py:78]: Oh, I see. Yes, indeed, `PyDatasetAdapter` shuffles batches, not samples.

So you can shuffle them the way you did.
